src/server/scripts/Spells/spell_druid.cpp

```diff
@@ -856,6 +861,45 @@ class spell_dru_lunar_inspiration : public AuraScript
     }
 };
 
+// 392315 - Luxuriant Soil
+class spell_dru_luxuriant_soil : public AuraScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_DRUID_REJUVENATION });
+    }
+
+    bool CheckProc(AuraEffect const* aurEff, ProcEventInfo& /*eventInfo*/)
+    {
+        return roll_chance_i(aurEff->GetAmount());
+    }
+
+    void HandleProc(AuraEffect* /*aurEff*/, ProcEventInfo& eventInfo)
+    {
+        Unit* target = GetTarget();
+
+        // Note: let's use the ProcSpell's max. range.
+        float spellRange = eventInfo.GetProcSpell()->GetSpellInfo()->GetMaxRange();
+
+        std::list<Unit*> targetList;
+        Trinity::AnyFriendlyUnitInObjectRangeCheck checker(target, target, spellRange);
+        Trinity::UnitListSearcher<Trinity::AnyFriendlyUnitInObjectRangeCheck> searcher(target, targetList, checker);
```

Comment:
If you only want one unit you can use UnitLastSearcher here

would look like this

```cpp
Unit* unit = nullptr;
Trinity::AnyFriendlyUnitInObjectRangeCheck checker(target, target, spellRange);
Trinity::UnitLastSearcher<Trinity::AnyFriendlyUnitInObjectRangeCheck> searcher(target, unit, checker);
Cell::VisitAllObjects(target, searcher, spellRange);
```